lib/alchemy/resource.rb

```diff
@@ -159,19 +170,24 @@ def editable_attributes
       attributes.reject { |h| restricted_attributes.map(&:to_s).include?(h[:name].to_s) }
     end
 
-    # Returns all columns that are searchable
+    # Returns all attribute names that are searchable in the admin interface
     #
-    def searchable_attributes
-      attributes.select { |a| searchable_attribute?(a) }
-        .concat(searchable_relation_attributes?(attributes))
+    def searchable_attribute_names
+      if model.respond_to?(:searchable_alchemy_attributes)
+        model.searchable_alchemy_attributes
+      else
+        attributes.select { |a| searchable_attribute?(a) }
+          .concat(searchable_relation_attributes?(attributes))
```

Comment:
Just recognized that we introduced a wrong method name in the last pull request `searchable_relation_attributes?` should not have a question mark in the the method name, because it does not return a Boolean, it returns a list of attributes instead. Would you mind to add a commit that fixes this? That would be great.